src/main/java/httpserver/HttpParser.java

```diff
@@ -1,13 +1,18 @@
 package httpserver;
 
 import java.util.Arrays;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 public class HttpParser implements Parseable{
     @Override
-    public String parse(String request) {
+    public Map<String, String> parse(String request) {
         List<String> splitRequest = Arrays.asList(request.split(System.lineSeparator()));
         List<String> splitRequestLine = Arrays.asList(splitRequest.get(0).split(" "));
-        return splitRequestLine.get(1);
+        Map<String, String> parsedRequest = new HashMap<>();
+        parsedRequest.put("method", splitRequestLine.get(0));
+        parsedRequest.put("path", splitRequestLine.get(1));
+        return parsedRequest;
```

Comment:
What happens when the string can't be parsed?

Reply:
Shouldn't an http request always have a request line?

Reply:
Sure - but what if I send a line that's invalid. instead of "GET /blah" I send "supercalifragilisticexpialidocious" - right now it looks like you'd crash.

Reply:
Good point.  I'll modify.

Reply:
Currently if you enter something that can't be parsed, an exception is raised and the connection disconnected.  Do you think that behavior suffices? Or, do you think we need to provide an error response?

Reply:
Here's some information from the RFC html document:  a recipient MAY attempt to recover a usable
   protocol element from an invalid construct.  HTTP does not define
   specific error handling mechanisms except when they have a direct
   impact on security, since different applications of the protocol
   require different error handling strategies.  For example, a Web
   browser might wish to transparently recover from a response where the
   Location header field doesn't parse according to the ABNF, whereas a
   systems control client might consider any form of error recovery to
   be dangerous.

Reply:
Recipients of an invalid request-line SHOULD respond with either a
   400 (Bad Request) error or a 301 (Moved Permanently) redirect with
   the request-target properly encoded.

Reply:
Considering the above, I think handling the error by responding with a 400 bad request might work instead of just throwing an exception.  Thoughts?

Reply:
That sounds correct to me - you'll want to send the 400 (I don't know why you'd do a 301, spec or not, that doesn't make sense to me). Plus of course you want to make sure you don't crash the server.